mathics/builtin/box/inout.py

```diff
@@ -465,41 +485,41 @@ def apply_list(self, boxes, evaluation):
         return result
 
     def init(self, *items, **kwargs):
-        # TODO: check that each element is an string or a BoxConstruct
+        # TODO: check that each element is an string or a BoxExpression
         self.box_options = {}
         if isinstance(items[0], Expression):
             if len(items) != 1:
                 raise Exception(
-                    items, "is not a List[] or a list of Strings or BoxConstruct"
+                    items, "is not a List[] or a list of Strings or BoxExpression"
                 )
             if items[0].has_form("List", None):
                 items = items[0]._elements
             else:
                 raise Exception(
-                    items, "is not a List[] or a list of Strings or BoxConstruct"
+                    items, "is not a List[] or a list of Strings or BoxExpression"
                 )
-        for item in items:
-            if not isinstance(item, BoxConstruct):
+
+        def check_item(item):
+            if isinstance(item, String):
+                return _BoxedString(item.value)
+            if not isinstance(item, BoxExpression):
                 raise Exception(
-                    item, "is not a List[] or a list of Strings or BoxConstruct"
+                    item, "is not a List[] or a list of Strings or BoxExpression"
                 )
-        self.items = tuple(items)
+            return item
 
-        elements = []
-        for item in items:
-            if isinstance(item, BoxConstruct):
-                elements.append(item.to_expression())
-            else:
-                elements.append(item)
-        elements = Expression(SymbolList, *elements)
-        self._elements = tuple((elements,))
+        self.items = tuple((check_item(item) for item in items))
+        self._elements = None
 
     def to_expression(self):
         """
         returns an evaluable expression.
```

Comment:
I meant to mention this from the #311. Please let us make the same change were we use the simpler more natural `->Expression` rather than gobbledigook. 

However, where a comment would be most appreciated is in line 522 with something along the lines of why we need to use `Expression()` and not `RowBox()` so that someone (probably me) in the future doesn't get confused again. 

Also, if this is the merge of ListExpression "lite", then ListExpression(*items) ot `to_mathics_list()` neither work?